mockingbird-compiler/src/main/kotlin/com/careem/mockingbird/MockingbirdPlugin.kt

```diff
@@ -15,116 +15,37 @@
  */
 package com.careem.mockingbird
 
-import com.squareup.kotlinpoet.metadata.KotlinPoetMetadataPreview
-import com.squareup.kotlinpoet.metadata.toKmClass
-import kotlinx.metadata.KmClass
 import org.gradle.api.Plugin
 import org.gradle.api.Project
 import org.gradle.api.logging.Logger
 import org.gradle.api.logging.Logging
 import org.gradle.kotlin.dsl.add
 import org.gradle.kotlin.dsl.get
-import org.jetbrains.kotlin.gradle.dsl.KotlinCompile
-import org.jetbrains.kotlin.gradle.dsl.KotlinMultiplatformExtension
-import java.io.File
 
-private const val EXTENSION_NAME = "mockingBird"
+internal const val EXTENSION_NAME = "mockingBird"
 
-
-@Suppress("UnstableApiUsage")
-@KotlinPoetMetadataPreview
 abstract class MockingbirdPlugin : Plugin<Project> {
 
-    private lateinit var classLoader: ClassLoaderWrapper
-    private lateinit var functionsMiner: FunctionsMiner
-    private lateinit var projectExplorer: ProjectExplorer
-    private lateinit var mockGenerator: MockGenerator
-    private val logger: Logger = Logging.getLogger(this::class.java)
+    private val mockingbirdPluginLegacyCodeGenDelegate = MockingbirdPluginLegacyCodeGenDelegate()
+    private val mockingbirdPluginKspDelegate = MockingbirdPluginKspDelegate()
 
-    private fun setupDependencies(target: Project) {
-        classLoader = ClassLoaderWrapper(projectExplorer, target)
-        functionsMiner = FunctionsMiner(classLoader)
-        mockGenerator = MockGenerator(classLoader, functionsMiner)
-    }
+    private val logger: Logger = Logging.getLogger(this::class.java)
 
     override fun apply(target: Project) {
-        val sourceSetResolver = SourceSetResolver()
```

Comment:
all the old plugin logic has been moved to `MockingbirdPluginLegacyCodeGenDelegate` without any change